src/app/reporting/ReportSchedulerImpl.cpp

```diff
@@ -144,24 +156,23 @@ bool ReportSchedulerImpl::IsReportScheduled(ReadHandler * aReadHandler)
 CHIP_ERROR ReportSchedulerImpl::CalculateNextReportTimeout(Timeout & timeout, ReadHandlerNode * aNode)
 {
     VerifyOrReturnError(nullptr != FindReadHandlerNode(aNode->GetReadHandler()), CHIP_ERROR_INVALID_ARGUMENT);
-    Timestamp now = mTimerDelegate->GetCurrentMonotonicTimestamp();
 
     // If the handler is reportable now, just schedule a report immediately
-    if (aNode->IsReportableNow())
+    if (aNode->IsReportableNow(mNow))
```

Comment:
So is the reason to have mNow just to avoid having a `now` argument to `CalculateNextReportTimeout` and `ScheduleReport`?

The need to update the `mNow` in some places _but not others_ is very fragile, it seems to me.  It would be better to have things take a `now` argument if they don't expect to compute it themselves and just have stack `now` at the entrypoints that do compute it.

Reply:
Replaced this by stack `nows`